packages/eas-cli/src/commands/env/push.ts

```diff
@@ -0,0 +1,171 @@
+      if (
+        existingVariableUpdate &&
+        existingVariableUpdate.value !== variable.value &&
+        variable.value !== null
+      ) {
+        existingDifferentVariables.push(variable);
+      } else {
+        delete updateVariables[variable.name];
+      }
+    });
+
+    const existingDifferentSharedVariables = existingDifferentVariables.filter(
+      variable => variable.scope === EnvironmentVariableScope.Shared
+    );
+
+    if (existingDifferentSharedVariables.length > 0) {
+      const existingDifferentSharedVariablesNames = existingDifferentSharedVariables.map(
+        variable => variable.name
+      );
+      Log.error('Shared variables cannot be overwritten.');
+      Log.error('Remove them from the env file or unlink them from the project to continue:');
+      existingDifferentSharedVariablesNames.forEach(name => Log.error(`- ${name}`));
+      throw new Error('Shared variables cannot be overwritten by eas env:push command');
+    }
+
+    if (existingDifferentVariables.length > 0) {
+      Log.warn('Some variables already exist in the environment.');
+
+      const { variablesToOverwrite } = await promptAsync({
+        type: 'multiselect',
+        name: 'variablesToOverwrite',
+        message: 'Select variables to overwrite:',
+        // @ts-expect-error property missing from `@types/prompts`
+        optionsPerPage: 20,
+        choices: existingDifferentVariables.map(variable => ({
+          title: `${variable.name}: ${updateVariables[variable.name].value}`,
+          value: variable.name,
+        })),
+      });
+
+      for (const existingVariable of existingVariables) {
+        const name = existingVariable.name;
+        if (variablesToOverwrite.includes(name)) {
+          updateVariables[name]['overwrite'] = true;
+        } else {
+          delete updateVariables[name];
+        }
+      }
+    }
+
+    const variablesToPush = Object.values(updateVariables);
+
+    if (variablesToPush.length === 0) {
+      Log.log('No new variables to push.');
+      return;
+    }
+
+    await EnvironmentVariableMutation.createBulkEnvironmentVariablesForAppAsync(
+      graphqlClient,
+      variablesToPush,
+      projectId
+    );
+
+    Log.log(`Uploaded env file to ${environment} environment.`);
+  }
+
+  private async parseEnvFileAsync(
+    envPath: string,
+    environment: string
+  ): Promise<Record<string, EnvironmentVariablePushInput>> {
+    if (!(await fs.exists(envPath))) {
+      throw new Error(`File ${envPath} does not exist.`);
+    }
+    const pushInput: Record<string, EnvironmentVariablePushInput> = {};
+
+    const variables: Record<string, string> = dotenv.parse(await fs.readFile(envPath, 'utf8'));
+
+    for (const [name, value] of Object.entries(variables)) {
+      pushInput[name] = {
+        name,
+        value,
+        environment,
+        visibility: name.startsWith('EXPO_SECRET')
+          ? EnvironmentVariableVisibility.Sensitive
```

Comment:
This is a super cool idea! 👏 

I'm wondering whether it wouldn't be better to make it
```suggestion
        visibility: name.startsWith('EXPO_SENSITIVE')
          ? EnvironmentVariableVisibility.Sensitive
```
?